trainers/dataloader.py

```diff
@@ -135,6 +138,49 @@ def process(example):
             # if we fail, destroy the file
             os.removedirs(self.tokenized_data_path)
             raise SystemExit from exc
+        
+    def get_data(self, split="train"):
+        """
+        Get the data by the split - e.g. train or val
+        """
+        ## load the data
+        data = np.memmap(
+            os.path.join(self.tokenized_data_path, f"{split}.bin"),
+            dtype=np.uint16,
+            mode="r",
+        )
+
+        ## additional steps for loading BytePooling and Conversational dataloaders
+        if hasattr(self, 'loading_shapes') and self.loading_shapes[split] is None:
+            if isinstance(self, BytePoolingDataloader):
+                self.loading_shapes[split] = (len(data)// self.model_cfg.embedder.byte_context_window, self.model_cfg.embedder.byte_context_window)
+
+                ## reset the data
+                data = None
```

Comment:
we probs want to rework this in the future... these should be delegated to the subclassing dataloaders... add a todo for that please

Reply:
i.e. instead of if else the load data should be handled by each subclass of the dataloader

Reply:
i actually hate isinstance(self, ...) - that implies something has gone horribly wrong imo

Reply:
https://stackoverflow.com/questions/53104810/check-if-self-is-instance-of-subclass-in-python

Reply:
can, i'll add it so. And you're right, I think subclass is better. Initially compiled it so that the subclasses will look neater.